exercises/practice/protein-translation/.meta/example.asm

```diff
@@ -0,0 +1,74 @@
+section .data
+protein_names:
+   Methionine     db "Methionine", 0
+   Phenylalanine  db "Phenylalanine", 0
+   Leucine        db "Leucine", 0
+   Serine         db "Serine", 0
+   Tyrosine       db "Tyrosine", 0
+   Cysteine       db "Cysteine", 0
+   Tryptophan     db "Tryptophan", 0
+
+codon_map:
+   db "AUG", 0
+   db "UUU", 1, "UUC", 1
+   db "UUA", 2, "UUG", 2
+   db "UCU", 3, "UCC", 3, "UCA", 3, "UCG", 3
+   db "UAU", 4, "UAC", 4
+   db "UGU", 5, "UGC", 5
+   db "UGG", 6
+   db "UAA", 7, "UAG", 7, "UGA", 7
+   db 0
+
+protein_list: dq Methionine, Phenylalanine, Leucine, Serine, Tyrosine, Cysteine, Tryptophan, 0
+
+section .bss
+output: resq 10
+
+section .text
+
+global proteins
+proteins:                 ;;extern const char **proteins(const char *rna);
+   cld
+   lea rdx, [rel output]
+   xor eax, eax           ; clear upper bits
+.next_codon:
+   mov rbx, rdi           ; after call to find_codon, rdi points to the next codon
```

Comment:
We should be saving rbx, and restoring the original value before we return.

https://wiki.osdev.org/System_V_ABI#x86-64
"Functions preserve the registers rbx, rsp, rbp, r12, r13, r14, and r15"

[Wikipedia - System V AMD64 ABI](https://en.wikipedia.org/wiki/X86_calling_conventions)
"If the callee wishes to use registers RBX, RSP, RBP, and R12–R15, it must restore their original values before returning control to the caller."

At the start:
`push rbx`

Before return:
`pop rbx`

Another option would be to not use rbx, and use r8 instead.